### DISCUSSION TITLE
The ultralytics training cannot be started normally.

### DISCUSSION THREAD

**Discussion by jane000** [ultralytics/ultralytics#22112] [Q&A] [answered]
I followed the official guidance of ultralytics for image training (30 images), but the training process always got stuck during the scanning stage. After a period of time, it automatically stopped. The training did not start properly. May I ask how I can solve this problem? It has been bothering me for several days. (Environment: CUDA13.0, Pytorch(supporting 12.9), python 3.12, VScode)

**Comment by jane000** [comment]
VScode Terminal Output：   
Transferred 319/355 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
train: Fast image access  (ping: 0.20.0 ms, read: 257.5163.9 MB/s, size: 125.8 KB)
train: Scanning D:\yolo\yolo_train_test\datasets\phone\labels.cache... 30 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 30/30  0.0s
val: Fast image access  (ping: 0.30.1 ms, read: 261.8218.9 MB/s, size: 162.6 KB)
val: Scanning D:\yolo\yolo_train_test\datasets\phone\labels.cache... 30 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 30/30 47.3Kit/s 0.0s
PS D:\yolo\yolo_train_test>

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jane000, thank you for your interest in Ultralytics 🚀! This is an automated response to help get things moving quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. For your case (training stuck during dataset scan), please include:
- The exact command(s) you ran (CLI or Python)
- A small sample of your dataset or a dummy dataset that reproduces the issue
- Your dataset YAML (redact any sensitive paths if needed)
- Full console logs with any warnings/errors
- Output of:
```bash
python -c "import ultralytics, torch, sys; print('ultralytics', ultralytics.__version__); print('torch', torch.__version__); print('python', sys.version)"
```
- Your OS/GPU details and whether this occurs in a clean environment

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by jane000** [reply]
It's a pity that this method didn't work

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

And also the training command you used.

**Reply by jane000** [reply, marked as answer]
Thank U . I  have used elder version Pytorch to solve it.

**Reply by glenn-jocher** [reply]
Thanks for the update—glad it’s working now. Stalls at scan are often Torch/CUDA mismatches, so for stability please pin a compatible torch/torchvision pair and verify with our torch/torchvision compatibility check, and when updating follow the Install Ultralytics guide (install PyTorch first). If everything looks good, please mark this discussion as answered. See: [torch/torchvision compatibility check](https://docs.ultralytics.com/reference/utils/checks/#ultralytics.utils.checks.check_torchvision), [Install Ultralytics guide](https://docs.ultralytics.com/quickstart/).
